### PR TITLE
GUI for related holdings

### PR DESCRIPTION
**Pull request by petergarnaes**
The site now has tabs for its different views of the data that is searched and selected on the left part of the screen.

The new tab is for exploring related holdings to the bibliographic post. The index key explorer from the previous GUI is reused for exploring the index keys of related holdings by making the index key explorer a higher order component.

Everything specific to the explorer of the index keys of bibliographic posts are moved to 'bibliographic_explorer', and now there is also a 'manifestation_explorer'.

New endpoint for pulling related holdings added.

Enzyme testing added, although not very comprehensive. I would rather make a separate PR with comprehensive integration testing involving all the components, which would have made this PR even larger and messier.

### CODE REVIEW

File: gui/app/reducers/related_holdings.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            let validAction = !!(action.item) &&
+                (action.item.bibliographicRecordId !== undefined) &&
+                (action.item.agencyId !== undefined);
+            action.item = (action.item) ? action.item : {
+                bibliographicRecordId: null,
+                agencyId: null
+            };
+            let actionBibId = action.item.bibliographicRecordId;
+            let actionAgencyId = action.item.agencyId;
+            // If values are undefined, set to null instead
```
Comment (petergarnaes) [inline gui/app/reducers/related_holdings.js]: Implementeret input validering i reducer, ville også gøre det for de andre reducere i det store test PR med UI integration tests og forbedringer til de eksisterende ruder tests.

File: service/src/main/java/dk/dbc/search/solrdocstore/FrontendAPIBean.java [hunk 2]
```diff
@@ -41,7 +39,33 @@ public Response getBibliographicKeys(@PathParam("bibliographicRecordId") String
...
+        log.info("Requesting bibliographic record id: {} and bibliographic agency id: {}",
+                bibliographicRecordId,bibliographicAgencyId);
+        TypedQuery<HoldingsItemEntity> query = entityManager.createQuery("SELECT h " +
+                "FROM HoldingsItemEntity h " +
+                "WHERE h.agencyId IN (" +
+                "SELECT h2b.agencyId " +
+                "FROM HoldingsToBibliographicEntity h2b " +
+                "WHERE h2b.bibliographicRecordId = :bibId AND h2b.bibliographicAgencyId = :agId)",
+                HoldingsItemEntity.class);
+        query.setParameter("bibId",bibliographicRecordId);
```
Comment (petergarnaes) [inline service/src/main/java/dk/dbc/search/solrdocstore/FrontendAPIBean.java]: Er dette query performant nok? Eller er det lidt ligemeget?

File: service/src/main/java/dk/dbc/search/solrdocstore/FrontendAPIBean.java [hunk 3]
```diff
@@ -41,7 +39,33 @@ public Response getBibliographicKeys(@PathParam("bibliographicRecordId") String
...
 
+    /**
+     * Returns a json object with a result field, which is a list of json HoldingsItemEntity mapped via the
+     * holdingsToBibliographic table.
+     * @param bibliographicRecordId
+     * @param bibliographicAgencyId
+     * @return Response
+     */
+    @GET
+    @Path("getRelatedHoldings/{bibliographicRecordId}&{bibliographicAgencyId}")
```
Comment (petergarnaes) [inline service/src/main/java/dk/dbc/search/solrdocstore/FrontendAPIBean.java]: Skal jeg lave det om til traditionelle URL parametre alla: 'bibliographicRecordId={bibliographicRecordId}&bibliograhpicAgencyId={bibliograhpicAgencyId}'? Er det konventionen?

**Comment by petergarnaes** [comment]
Hmm... Merge build virkede fint tidligere, tror commit 41108a9e7b7868e4e01abe1fee2847c88b2088bd er synderen...

Fra log:
14:19:53 [INFO] solr-doc-store-corepo-worker ....................... FAILURE [06:13 min]

Har ikke rørt Corepo worker.

File: service/src/main/java/dk/dbc/search/solrdocstore/FrontendAPIBean.java (hunk 2 above)
Comment (ja7) [reply service/src/main/java/dk/dbc/search/solrdocstore/FrontendAPIBean.java]: Der er index på så det kan ikke være meget bedre. 

```SQL
CREATE INDEX holdingsToBibliographic_Bibliographic
    ON holdingsToBibliographic (bibliographicRecordId, bibliographicAgencyId);
```

File: gui/app/api/index.js [hunk 4]
```diff
@@ -1,14 +1,20 @@
+let parse = (res) => {
+    if(res.status === 200)
+        return res.json();
+    else
+    if(res.status === 400)
+        throw new Error("Input error, server failed to URL decode bibliographicRecordId");
```
Comment (ja7) [inline gui/app/api/index.js]: Da parse nu håndtere flere forskellige requests  så er en status 400 nok ikke lig med 
server failed to URL decode bibliographicRecordId. 
Der udover så vil det nok være rart hvis Fejl texten fra serveren kan ses et sted.

File: service/src/main/java/dk/dbc/search/solrdocstore/FrontendAPIBean.java (hunk 3 above)
Comment (ja7) [reply service/src/main/java/dk/dbc/search/solrdocstore/FrontendAPIBean.java]: Tænker det enten skal være traditioneller url parametre eller også med / som path seperator 
Med path Seperator vil jeg nok mene at  Agency skal være før bibliographicRecordId 
Så 
@Path("getRelatedHoldings/{bibliographicAgencyId}/{bibliographicRecordId}")

**Pull request merged**